granulate_utils/linux/cgroups/base_cgroup.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+import os
+from enum import Enum
+from pathlib import Path
+from typing import List
+
+from granulate_utils.linux.cgroups.exceptions import MissingCgroup, MissingController, SkippedCgroup, UnsupportedCgroup
+
+PID_CGROUPS = Path("/proc/self/cgroup")
+CGROUP_PARENT_PATH = Path("/sys/fs/cgroup")
+
+
+class HIERARCHIES(Enum):
+    memory = "memory"
+    cpu = "cpu,cpuacct"
+
+
+class CgroupVerifications:
+    @staticmethod
+    def cgroup_v1() -> None:
+        if len(PID_CGROUPS.read_text().split("\n")) == 2:
+            raise UnsupportedCgroup("version 2")
+
+    @staticmethod
+    def supported_cgroup(cgroup_type: str) -> None:
+        assert cgroup_type, "must provide cgroup type"
+        if cgroup_type not in HIERARCHIES.__members__:
+            raise UnsupportedCgroup(cgroup_type)
+
+    @staticmethod
+    def ignored_cgroup(cgroup: str, cgroup_name: str) -> None:
+        ignore_list = ["kubepods", "docker", "ecs"]
+        if any(x in ignore_list for x in cgroup.split("/")):
+            raise SkippedCgroup(cgroup, cgroup_name)
+
+    @staticmethod
+    def controller_exists(controller_path: Path) -> None:
+        if not controller_path.is_file():
+            raise MissingController(controller_path)
+
+
+class BaseCgroup:
+    HIERARCHY = ""
+
+    def __init__(self) -> None:
+        CgroupVerifications.cgroup_v1()
+        CgroupVerifications.supported_cgroup(self.HIERARCHY)
+
+    def _get_cgroup(self) -> str:
```

Comment:
This is practically the existing `get_cgroups` function, why not use it

Reply:
I disagree - well, it's more important to have logic written in one site and not duplicated. You can write a function here `_get_cgroup` that calls the other one. You can change the other one to call this function. I don't mind, but don't duplicate logic